while-loops/while-loops.md

```diff
@@ -0,0 +1,257 @@
+
+## While Loops Syntax
+
+While loops are loops that run _while_ some condition is `True` or truthy.
+
+``` python
+while conditional_evaluates_to_true:
+    print("I'm in the loop body")
+```
+
+| Piece of code                  | Notes                                                                                                                                              |
+|--------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|
+| `while` | `while` is a keyword in Python that begins a while loop. It looks immediately to the right for a conditional expression, and a `:` to end the line |
+| `conditional_evalutes_to_true` | **Replace this** with a conditional expression. When this expression is true, the loop body will run                                               |
+| Loop body                      | **Replace this** with the code that should run in this while loop. This should be indented once inside the while loop.                             |
+
+### Practicing Reading While Loops
+
+Here, we use a while loop to print the numbers 1 through 5.
+
+``` python
+i = 1
+while i <= 5:
+    print(f"The value of i is {i}")
+    i += 1
+
+print("We are outside the while loop! Wooo!")
+```
+
+Here, we have a variable `i` . While `i` is less than or equal to 5, we will print `f"The value of i is {i}"` . Then, we will increment `i` by one.
+
+`i` begins with a value of `1` . As the while loop loops, `i` grows bigger and bigger, since we're modifying the same `i` variable.
+
+At some point, `i` will have a value of `6` . Then, `i <= 5` is falsey, and the while loop will exit. Our code execution moves on to the final `print` statement.
+
+Another way to read this while loop is with a table:
+
+| Value of `i` at the beginning of the loop | Is `i <= 10` ? | Value of `i` at the end of the loop body |
+|-------------------------------------------|---------------|------------------------------------------|
+| `1` | `True` | `2` |
+| `2` | `True` | `3` |
+| `3` | `True` | `4` |
+| `4` | `True` | `5` |
+| `5` | `True` | `6` |
+| `6` | `False` | Loop body doesn't execute                |
+
+Here, we should observe that at some point, `i` has a value of `6` , and the code asked "Is 6 <= 5? Is that true?" Because the answer is false, the code skips over the rest of the loop.
+
+Callout: While loops can accidentally run infinitely! If the conditional for the while loop never becomes false, then the while loop and the program keep going, and going, and going... until something forces it to stop (such as a human closing down the program), or our machine cannot run it anymore. We call these cases **infinite loops, ** and are always a bug.
```

Comment:
Markdown formatting for bolding isn't working, maybe should be `**infinite loops**,` with the comma outside?

> and are always a bug

Recommend rewording to "and are often a bug".

There are plenty of examples of programs with infinite loops that are not bugs. Event loop based concurrency is often implemented with an explicit `while True:` loop. E.g. https://tutorialedge.net/python/concurrency/asyncio-event-loops-tutorial/

Reply:
Thank you for the accuracy callout! I totally agree.

```suggestion
Callout: While loops can accidentally run infinitely! If the conditional for the while loop never becomes false, then the while loop and the program keep going, and going, and going... until something forces it to stop (such as a human closing down the program), or our machine cannot run it anymore. We call these cases **infinite loops,** and are often a bug.
```

Fairly confident the formatting mistake was because of the leading space